infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -120,6 +125,47 @@ public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
         if (Math.max(getGlobalTail(), 0L) < getTrimMark()) {
             syncTailSegment(getTrimMark() - 1);
         }
+
+        // Initialize stream tails
+        initializeStreamTails();
+    }
+
+    private void initializeStreamTails() {
+        long firstSegment = 0;
+        long lastSegment = globalTail.get();
+
```

Comment:
This doesnt account for segments that have been prefix trimmed. Also, globalTail is not the last segment.

Reply:
done.